dataset/include/scipp/dataset/bin.h

```diff
@@ -8,10 +8,9 @@
 
 namespace scipp::dataset {
 
-SCIPP_DATASET_EXPORT DataArray sortby(const DataArrayConstView &data,
-                                      const Dim dim);
 SCIPP_DATASET_EXPORT DataArray
-bucketby(const DataArrayConstView &array,
-         const std::vector<VariableConstView> &edges);
+bin(const DataArrayConstView &array,
```

Comment:
Is it possible to bin a Dataset or Variable if I wish to?

Reply:
Not a variable, since you alsways need a coordinate to find the target bin. We could in principle add support for datasets, since they have coords, but since it has limited use I did not, for now. It would probably just be a wrapper that bins all dataset entries.

Reply:
How do you create a Variable which has a `dtype=dtype<bucket<Variable>`?

Reply:
`binned = sc.bins(begin, end, dim, var)`, see, e.g., https://github.com/scipp/scipp/blob/f779c499bbf3e1b8a88c55e5084bc922907b0cdf/python/tests/test_bins.py#L11-L13.

Reply:
Thanks. Is the key differance between `bin` and `bins` that one defines bins by groups/edges and the other defines them by supplying the lower level index pairs?

As a future note having `sc.bin` and `sc.bins` doing subtly different things feels potentially a bit confusing from an API perspective?

Reply:
You are not the first to express this opinion. If have argued that it is not dangerous, since keyword args are different. Not sure what the ultimate conclusion is.